include/zephyr/init.h

```diff
@@ -205,7 +205,7 @@ struct init_entry {
 #define SYS_INIT_NAMED(name, init_fn_, level, prio)                                       \
 	static const Z_DECL_ALIGN(struct init_entry)                                      \
 		Z_INIT_ENTRY_SECTION(level, prio, 0) __used __noasan                      \
-		Z_INIT_ENTRY_NAME(name) = {.init_fn = {.sys = (init_fn_)}, .dev = NULL}
```

Comment:
would `{.init_fn = {.sys = (init_fn_)}, .dev = {NULL}}` not work? not sure why are removing the field names

Reply:
oh, union with no name

Reply:
how about

```diff
-               Z_INIT_ENTRY_NAME(name) = {.init_fn = {.sys = (init_fn_)}, .dev = NULL}
+               Z_INIT_ENTRY_NAME(name) = {.init_fn = {.sys = (init_fn_)}, {.dev = NULL}}
```

?